sonic_platform_base/chassis_base.py

```diff
@@ -152,6 +158,45 @@ def get_component(self, index):
 
         return component
 
+    def get_all_power_consumers(self):
+        """
+        Retrieves a list of all power consumers in a modular chassis
+
+        Returns:
+            A list, where each entry is a unique power consumer. For example,
+            'FAN', 'CARD' etc
+        """
+        return None
+
+    def get_total_consumer_power(self, consumer):
+        """
+        Retrives the maximum total power drawn by the consumer component
+
+        Returns:
+            A float, with value of the maximum consumable power of the
+            component.
+        """
+        return 0.0
+
+    def get_status_master_led(self, device_type):
```

Comment:
There seems to be an overlap with https://github.com/Azure/sonic-platform-common/commit/9f3be4446f708d5a171b1be530a980ef2e822838

Reply:
This one is for master led. 
per PSU led is set by 9f3be44

Reply:
As per the discussion I would be advocating for the following to manage status led
```
Chassis.get_status_led(self, name)
```

This way we could agree on a common name for the `master_led` and psud could change if if necessary.

```
led = chassis.get_status_led(chassis.LED_MASTER)
if led:
    led.set_color(chassis.COLOR_AMBER)
```

All other status led would also benefit from the same feature set and adapt to various platform designs.
Given that platforms might or might not have the same leds, platform daemon could conditionally set the leds when they exist.

@jleveque do you have some thoughts?

Reply:
@Staphylo: I think your idea is reasonable.

@Junchao-Mellanox: How does this sound to you?

Reply:
From my understanding, if user wants to get/set PSU/FAN led, you could do it like:

```
led_color = chassis.get_psu(psu_index).get_status_led()
chassis.get_psu(psu_index).set_status_led("red")
led_color = chassis.get_fan(fan_index).get_status_led()
chassis.get_psu(fan_index).set_status_led("green")
```

So I would like to understand, on which circumstances that we need the new API get_status_master_led/set_status_master_led.

BTW the method "get_status_led" returns the led color instead of a led object, we don't have led object defined in platform API for now.

Reply:
These are applicable for the modular chassis. For fixed platform I can draw a parallel to system led. The master led will indicate comprehensive state of each type of peripheral.

For example, master PSU led will indicate where the chassis is fully powered, is low or under powered. Similarly, there is a Master fabric led or fan led state. The individual leds on a chassis sometime could not be visible on the front of the chassis. The Master Led on the supervisor will provide these indicators.

Reply:
OK, understood. Then it looks good to me. As a PSU master led is supposed to indicate the PSU state, another option is to add the master led as a class level method in psu_base.py, fan_base.py and fan_drawer.py. Something like:

```python
@classmethod
def get_status_master_led(cls):
    ...
```

Reply:
@Junchao-Mellanox - The classmethod approach is good. However, the implementation will still be in Psu(PsuBase) class. I will not be able to implement in the base class. If I implement in Psu class, then I have to try and import Psu from sonic_platform.psu. 
Is my understanding correct?

Reply:
Yes, you are correct.

Reply:
Fixed. Please also see https://github.com/Azure/sonic-platform-daemons/pull/104 for usage.